daemon/api_system_secureboot.go

```diff
@@ -0,0 +1,151 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/fdestate"
+)
+
+var systemSecurebootCmd = &Command{
+	// TODO GET returning whether secure boot is relevant for the system?
+
+	Path: "/v2/system-secureboot",
+	POST: postSystemSecurebootAction,
+	WriteAccess: interfaceProviderRootAccess{
+		// TODO find a specialized interface for this, but for now assume that
+		// requests will come only from snaps plugging fwupd interface on the
+		// slot side, which also allows manipulation of EFI variables
+		Interfaces: []string{"fwupd"},
+	},
+}
+
+func postSystemSecurebootAction(c *Command, r *http.Request, user *auth.UserState) Response {
+	contentType := r.Header.Get("Content-Type")
+
+	switch contentType {
+	case "application/json":
+		return postSystemSecurebootActionJSON(c, r)
+	default:
+		return BadRequest("unexpected content type: %q", contentType)
+	}
+}
+
+func postSystemSecurebootActionJSON(c *Command, r *http.Request) Response {
+	var req struct {
+		Action string           `json:"action,omitempty"`
+		Data   *json.RawMessage `json:"data"`
```

Comment:
I would avoid having this nested "data" level, just define a top level struct with the field we need with a validate method that checks if field that should be set for a given op are or than't shouldn't aren't

Reply:
Updated